### PR TITLE
Fix bad xrefs

### PR DESCRIPTION
**Pull request by gewarren**
Suggest ignoring whitespace changes in the diff. The build is clean for these files so the xrefs are good.

Changes:

- remove extraneous preceding characters, for example `[\], Type\[\], Type\[\], Type\[\]\[\], Type\[\]\<xref:System.Reflection.Emit.TypeBuilder.DefinePInvokeMethod%2A>`
- replace %28 with `(`, %29 with `)`, %2C with `,` (and a few others to make xrefs more readable)

Contributes to https://github.com/dotnet/docs/issues/19387.
@mairaw Thanks for letting me know about these!

### CODE REVIEW

**Comment by mairaw** [comment]
No problem! I know some of these links were not being reported on the build report but I don't see any drops on the number of warnings. Is that expected for this particular PR @gewarren?

File: xml/System/_AppDomain.xml [hunk 1]
```diff
@@ -1785,14 +1785,6 @@
         <param name="rawSymbolStore">An array of type <see langword="byte" /> containing the raw bytes representing the symbols for the assembly.</param>
         <summary>Provides COM objects with version-independent access to the <see cref="M:System.AppDomain.Load(System.Byte[],System.Byte[])" /> method overload.</summary>
         <returns>The loaded assembly.</returns>
-        <remarks>
-          <format type="text/markdown"><![CDATA[  
-  
-## Remarks
```
Comment (gewarren) [inline xml/System/_AppDomain.xml:1791]: This just seemed to point to itself, so I deleted it.

**Comment by gewarren** [comment]
> No problem! I know some of these links were not being reported on the build report but I don't see any drops on the number of warnings. Is that expected for this particular PR @gewarren?

Yes, the "xref links" weren't previously rendering as links when they were in the form `[\], Type\[\], Type\[\], Type\[\]\[\], Type\[\]\<xref:System.Reflection.Emit.TypeBuilder.DefinePInvokeMethod%2A>`. So they didn't generate any warnings. Now that they are links, they didn't introduce any new warnings either.

**Comment by gewarren** [comment]
Thanks for the reviews Maira and Bill!

**Pull request merged**